servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java

```diff
@@ -4770,6 +4792,14 @@ public Assertion putAssertion(ResourceContext ctx, String domainName, String pol
 
         validatePolicyAssertion(assertion, caller);
 
+        // we are not going to allow any user to add/update
+        // an assertion that the associated role does not exist
+
+        RoleList roleList = getRoleList(ctx, domainName, null, null);
```

Comment:
Same comment here as above. getRoleList is an API endpoint. Let's replace this with an internal implementation and not call another API endpoint.